aptos-move/framework/src/natives/function_info.rs

```diff
@@ -127,7 +127,7 @@ fn native_check_dispatch_type_compatibility_impl(
         rhs.ty_param_abilities() == lhs.ty_param_abilities()
             && rhs.return_tys() == lhs.return_tys()
             && &lhs.param_tys()[0..lhs.param_count() - 1] == rhs.param_tys()
-            && !rhs.is_friend_or_private()
+            && rhs.is_public()
```

Comment:
Haven't looked at the context where this is used carefully, but do we have to consider `#[persistent]` functions here?

Reply:
These were added before `#[persistent]`, technically we can allow persistent here I think, but then this needs to be feature gated. In any case for actual closures we use persistent which is good, I am not sure if we need to allow this for native dispatch, cc @wrwg @runtian-zhou